### ISSUE TITLE
Object based notifications missing oldValue

### ISSUE THREAD

**Issue by rscholtes** [realm/realm-cocoa#4748]
## Goals
Retrieve correct object change notifications

## Expected Results

Get the oldValue and newValue properties set to the correct values

## Actual Results

oldValue is always nil

## Steps to Reproduce

Run the provided example project provided via e-mail

## Version of Realm and Tooling

Realm version: 2.4.4

Xcode version: 8.2.1

iOS/OSX version: 10.12.3

Dependency manager + version:  CocoaPods 1.2.1.beta.1

**Comment by jpsim** [comment]
Thanks for filing this @rscholtes. I was able to reproduce the issue with the project you sent. As far as I understand this, this would be a bug. But I'd like @tgoyne to confirm in case there's something else at play here.

**Comment by rscholtes** [comment]
Thanks for the quick reply! If you require anything else, let me know

**Comment by tgoyne** [comment]
[`oldValue` is not set if the write is made on the same thread as the notification](https://realm.io/docs/swift/latest/api/Structs/PropertyChange.html#/s:vV10RealmSwift14PropertyChange8oldValueGSqP__) and in your sample project all of the writes are being made on the main thread.

**Comment by jpsim** [comment]
Right. Since I don't think we have a ticket tracking lifting that limitation, I'll convert this ticket to a feature request, although I don't think this is something we'll be doing in the near future.

**Comment by hien352911** [comment]
Perhaps you are testing by changing the value directly in the code. I have tested the same. The old value is always nil. But if the value is changed at the realm studio or view controller, old value is always correct.
